File: core/SyncPlatform/src/Services/Payment/Payment.Domain/Models/UserSubscription.cs
using Libs.Shared.Common;
using Payment.Domain.Enums;

namespace Payment.Domain.Models;

public class UserSubscription : BaseAuditableEntity
{
    public Guid UserId { get; set; }

    public Guid SubscriptionPlanId { get; set; }

    public SubscriptionStatus Status { get; set; }

    public DateTimeOffset StartedAt { get; set; }

    public DateTimeOffset? ExpiredAt { get; set; }

    public bool AutoRenew { get; set; }

    public DateTimeOffset? LastBillingAt { get; set; }

    public DateTimeOffset? NextBillingAt { get; set; }

    public string? CancellationReason { get; set; }
}

File: ui/web/src/lib/validations/user-subscription.ts
import { z } from "zod";

export const userSubscriptionSchema = z.object({
  userId:                 z.string().min(1, "Required"),
  subscriptionPlanId:     z.string().min(1, "Required"),
  status:                 z.string().min(1, "Required"),
  startedAt:              z.string().min(1, "Required"),
  expiredAt:              z.string().optional(),
  autoRenew:              z.boolean().default(false),
  nextBillingAt:          z.string().optional(),
  cancellationReason:     z.string().max(512).optional(),
  managedBy:              z.string().min(1, "Required"),
  externalSubscriptionId: z.string().max(256).optional(),
});

export type UserSubscriptionFormValues = z.infer<typeof userSubscriptionSchema>;
